sdk/storage/src/blobs/blob_client.cpp

```diff
@@ -170,8 +171,40 @@ namespace Azure { namespace Storage { namespace Blobs {
     protocolLayerOptions.IfMatch = options.AccessConditions.IfMatch;
     protocolLayerOptions.IfNoneMatch = options.AccessConditions.IfNoneMatch;
 
-    return BlobRestClient::Blob::Download(
+    auto downloadResponse = BlobRestClient::Blob::Download(
         options.Context, *m_pipeline, m_blobUrl.ToString(), protocolLayerOptions);
+
+    {
+      // In case network failure during reading the body
+      std::string eTag = downloadResponse->ETag;
+
+      auto retryFunction
+          = [this, options, eTag](
+                const Azure::Core::Context& context,
+                const HTTPGetterInfo& retryInfo) -> std::unique_ptr<Azure::Core::Http::BodyStream> {
+        unused(context);
+
+        DownloadBlobOptions newOptions = options;
+        newOptions.Offset
+            = (options.Offset.HasValue() ? options.Offset.GetValue() : 0) + retryInfo.Offset;
+        newOptions.Length = options.Length;
+        if (newOptions.Length.HasValue())
+        {
+          newOptions.Length = options.Length.GetValue() - retryInfo.Offset;
+        }
+        if (!newOptions.AccessConditions.IfMatch.HasValue())
+        {
+          newOptions.AccessConditions.IfMatch = eTag;
+        }
+        return std::move(Download(newOptions)->BodyStream);
+      };
+
+      ReliableStreamOptions reliableStreamOptions;
+      reliableStreamOptions.MaxRetryRequests = 3;
```

Comment:
Hardcoded retry times is not ideal.

Reply:
So do you want it to be customizable or defined in a header?

Reply:
Although I really don't see why user would want to customize this value.

Reply:
Retry will happen only if Read() fails.  That means network errors mostly.
It is not related to retrying the request (that's would be the retry policy, which customer can configure).

Maybe if you happen to have some global config that customers can change before compiling, we could have it there.
Of it could be a compile option, maybe.
At any case, I don't think we need it configurable at runtime